web-api/src/personalization/routes.rs

```diff
@@ -532,7 +555,29 @@ async fn semantic_search(
     )
     .await?;
 
-    Ok(Json(SemanticSearchResponse {
+    if let Some(personalize) = personalize {
+        let interest = storage::Interest::get(&state.storage, &personalize.for_user).await?;
+        let tag_weights = storage::Tag::get(&state.storage, &personalize.for_user).await?;
+
+        if !interest.has_enough(state.config.as_ref()) {
+            return Ok(Either::Right((
+                Json(PersonalizedDocumentsError::NotEnoughInteractions),
+                StatusCode::CONFLICT,
+            )));
+        }
```

Comment:
:question: : It is not clear if this is the behavior we want or if we simple want to not personalize if there are not enough interactions.

Reply:
imo it's ok to fail in that case, i will ask in the daily about opinions on this.